### PR TITLE
Correctly free property and method lists

### PR DESCRIPTION
**Pull request by TitanNano**
While working on the patch for #554 I noticed that both `free_property_list_func` and `free_method_list_func` are not correctly freeing their data. 

## Property List
`free_property_list_func` still uses the `ScriptInstance` trait to obtain the list length. This is incorrect. It should take the length of the `Vec` which is being removed from the `HashMap`.

## Method List
Has the same issue as the property list but, moreover, it is also never dropping the `Vec` inside the `HashMap`. So this memory is effectively leaked.

### CODE REVIEW

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-670

File: godot-core/src/engine/script_instance.rs [hunk 1]
```diff
@@ -708,18 +703,23 @@ mod script_instance_info {
...
+        let vec = handle_panic(ctx, || {
             let instance = instance_data_as_script_instance::<T>(p_instance);
 
-            borrow_instance(instance).get_method_list().len()
+            instance
+                .method_list
+                .lock()
+                .expect("method_list mutext should not be poisoned")
+                .remove(&p_method_info)
+                .expect("godot is trying to free the method_list, but none has been set")
```
Comment (Bromeon) [inline godot-core/src/engine/script_instance.rs:714]:
```suggestion
                .expect("method_list mutex should not be poisoned")
                .remove(&p_method_info)
                .expect("Godot is trying to free the method_list, but none has been set")
```

Comment (Bromeon) [reply godot-core/src/engine/script_instance.rs:714]: \+ same above

File: godot-core/src/engine/script_instance.rs [hunk 2]
```diff
@@ -575,27 +575,22 @@ mod script_instance_info {
             )
         };
 
-        let length = handle_panic(ctx, || {
+        let vec = handle_panic(ctx, || {
```
Comment (Bromeon) [inline godot-core/src/engine/script_instance.rs:578]: Can this still panic? If yes, maybe consider commenting how.

File: godot-core/src/engine/script_instance.rs [hunk 3]
```diff
@@ -708,18 +703,23 @@ mod script_instance_info {
     ) {
         let ctx = || format!("error while calling {}::get_method_list", type_name::<T>());
 
-        let length = handle_panic(ctx, || {
+        let vec = handle_panic(ctx, || {
```
Comment (Bromeon) [inline godot-core/src/engine/script_instance.rs:706]: Also here, panic still applicable?

**Review by Bromeon — COMMENTED** [review]
Thank you!

File: godot-core/src/engine/script_instance.rs (hunk 1 above)
Comment (TitanNano) [reply godot-core/src/engine/script_instance.rs]: thanks.

File: godot-core/src/engine/script_instance.rs (hunk 2 above)
Comment (TitanNano) [reply godot-core/src/engine/script_instance.rs]: No, this should no longer panic.

**Comment by Bromeon** [comment]
Uh, CI again... will need to see if this is an issue that's also happening with other branches...

**Pull request merged**